frontend/appflowy_flutter/lib/user/application/notification_filter/notification_filter_bloc.dart

```diff
@@ -0,0 +1,75 @@
+import 'package:bloc/bloc.dart';
+import 'package:equatable/equatable.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'notification_filter_bloc.freezed.dart';
+
+class NotificationFilterBloc
+    extends Bloc<NotificationFilterEvent, NotificationFilterState> {
+  NotificationFilterBloc() : super(const NotificationFilterState()) {
+    on<NotificationFilterEvent>((event, emit) async {
+      await event.when(
+        update: (showUnreadsOnly, groupByDate, sortBy) async {
+          emit(
+            state.copyWith(
+              showUnreadsOnly: showUnreadsOnly,
+              groupByDate: groupByDate,
+              sortBy: sortBy,
+            ),
+          );
+        },
+        reset: () {
+          emit(const NotificationFilterState());
+        },
+      );
+    });
+  }
+}
+
+enum NotificationSortOption {
+  descending,
+  ascending,
+}
+
+@freezed
+class NotificationFilterEvent with _$NotificationFilterEvent {
+  const factory NotificationFilterEvent.update({
+    bool? showUnreadsOnly,
+    bool? groupByDate,
+    NotificationSortOption? sortBy,
+  }) = _Update;
```

Comment:
Should we consider splitting this interface into three separate interfaces?

- updateShow...
- updateGroup...
- updateSort...